host-bmc/host_pdr_handler.cpp

```diff
@@ -652,12 +652,62 @@ void HostPDRHandler::processHostPDRs(mctp_eid_t /*eid*/,
                 }
                 else
                 {
-                    rc = pldm_pdr_add_check(repo, pdr.data(), respCount, true,
-                                            pdrTerminusHandle, &rh);
-                    if (rc)
+                    if ((isHostPdrModified == true) || !(modifiedCounter == 0))
                     {
-                        // pldm_pdr_add() assert()ed on failure to add a PDR.
-                        throw std::runtime_error("Failed to add PDR");
+                        isHostPdrModified = false;
+
+                        pldm_delete_by_record_handle(repo, rh, true);
+
+                        pldm_pdr_add_check(repo, pdr.data(), respCount, true,
```

Comment:
do we want to handle rc from `pldm_pdr_add_check` here as well?

Reply:
done